src/escape.d

```diff
@@ -865,7 +867,16 @@ private void escapeByValue(Expression e, EscapeByResults* er)
             if (e.e1.op == TOKdotvar && t1.ty == Tfunction)
             {
                 DotVarExp dve = cast(DotVarExp)e.e1;
-                if (dve.var.storage_class & STCreturn || tf.isreturn)
+                FuncDeclaration fd = dve.var.isFuncDeclaration();
+                AggregateDeclaration ad;
+                if (global.params.vsafe && tf.isreturn && fd && (ad = fd.isThis()) !is null)
+                {
+                    if (ad.isClassDeclaration())       // this is 'return scope'
+                        dve.e1.accept(this);
```

Comment:
@WalterBright you may want to install the CodeCov Chrome extension, it nicely shows uncovered lines in red on this page!